Update test to reflect 403 status

ros3bud opened the pull request:
Addresses #8 

@zalary or @cgrayson, LGTY? 

It was recently discovered that we can get back a 403 error from webbula due to an invalid API key. This update to the test better reflects the way that only when the API key is correct and the Webbula server provides data about an email will the response be 200. The original response parsing still works because anything other than a 200 will return outcome:error.

cgrayson commented on spec/email_spec.coffee:
```diff
@@ -252,8 +252,8 @@ describe 'Error Response', ->
-        outcome: 'failure'
-        reason: '10111 : Invalid API Key'
+        outcome: 'error'
+        reason: 'Webbula error (403)'
```
But now the specific reason ("Invalid API Key") is lost. The status number gives a hint, but their error message spells it out so nicely. How about [looking for errors in the body](https://github.com/activeprospect/leadconduit-integration-webbula/blob/master/src/email.coffee#L57) in the non-200 case, too?

ros3bud replied:
@cgrayson - I see what you mean. I've updated the response method so that whenever an error is present it will be used. Otherwise the status will be used. Tests were also updated to reflect this change.

ros3bud commented:
- Updated tests to reflect updated parsing of billable flag.
- Updated the response parsing method for passing tests.
- Updated language in 1 unit test.

bhicks commented on src/email.coffee:
```diff
@@ -48,31 +48,31 @@ validate = (vars) ->
+  emailResult = emailsData?.result.toLowerCase()
+  webbulaError = parsedBody.errors?[0]
+
+  if webbulaError? or res.status !=200
```
I'm not sure of all the response statuses that they return, but it seems like we should only treat `4xx` or `5xx` responses as errors.

As a small side nitpick, there should be a space in `!=200`

ros3bud commented:
@bhicks - Webbula specifically gives back a 200 for all good requests, so anything that is not a 200 will contain an error response.

bhicks commented:
What would happen if they start to return other `2xx` response statuses for good requests, like `201` or `204`?

It seems like we could guard against this by checking for those status codes, something like:

```
if webbulaError? or res.status >= 400
```
